### PR TITLE
Do not perform xfs_repair on xfs filesystem

### PR DESCRIPTION
**Pull request by gnufied**
This partly reverts https://github.com/kubernetes/utils/pull/126 but still we perform filesystem repair just before mounting. 

fixes https://github.com/kubernetes/utils/issues/141

### CODE REVIEW

**Comment by gnufied** [comment]
cc @jsafrane @msau42

File: mount/mount_linux.go [hunk 1]
```diff
@@ -410,14 +383,7 @@ func (mounter *SafeFormatAndMount) formatAndMountSensitive(source string, target
...
 			// Run check tools on the disk to fix repairable issues, only do this for formatted volumes requested as rw.
-			var err error
-			switch existingFormat {
-			case "xfs":
-				err = mounter.checkAndRepairXfsFilesystem(source)
-			default:
-				err = mounter.checkAndRepairFilesystem(source)
-			}
-
+			err := mounter.checkAndRepairFilesystem(source)
```
Comment (gnufied) [inline mount/mount_linux.go]: code before change:

```
	if !readOnly {
		// Run fsck on the disk to fix repairable issues, only do this for volumes requested as rw.
		klog.V(4).Infof("Checking for issues with fsck on disk: %s", source)
		args := []string{"-a", source}
		out, err := mounter.Exec.Command("fsck", args...).CombinedOutput()
		if err != nil {
			ee, isExitError := err.(utilexec.ExitError)
			switch {
			case err == utilexec.ErrExecutableNotFound:
				klog.Warningf("'fsck' not found on system; continuing mount without running 'fsck'.")
			case isExitError && ee.ExitStatus() == fsckErrorsCorrected:
				klog.Infof("Device %s has errors which were corrected by fsck.", source)
			case isExitError && ee.ExitStatus() == fsckErrorsUncorrected:
				return NewMountError(HasFilesystemErrors, "'fsck' found errors on device %s but could not correct them: %s", source, string(out))
			case isExitError && ee.ExitStatus() > fsckErrorsUncorrected:
				klog.Infof("`fsck` error %s", string(out))
			}
		}
	}
```
and hence should be same as before.

Comment (saad-ali) [inline mount/mount_linux.go]: The code that is part of `checkAndRepairFilesystem(...)` used to be called before `// Check if the disk is already formatted`. As part of https://github.com/kubernetes/utils/pull/126 it was moved down here. Should we revert this change as well or is it necessary for some reason?

Comment (gnufied) [reply mount/mount_linux.go]: We could revert that change too but I figured it is harmless and helps to keep diff of this PR small. We will have to fix all the breaking tests because order in which `fsck` gets called will change. 

Also it is perfectly fine to *not* run `fsck` on a unformatted disk.

Comment (saad-ali) [reply mount/mount_linux.go]: Ack. Sounds good to me. @msau42 @xing-yang @jsafrane any objections?

Comment (xing-yang) [reply mount/mount_linux.go]: Sounds good to me.

**Comment by dims** [comment]
/approve
/hold

@xing-yang @saad-ali (others!) feel free to remove hold when this is ready

File: mount/mount_linux.go (hunk 1 above)
Comment (msau42) [reply mount/mount_linux.go]: lgtm. The bug report had the error past the fs format check, so it's not the fs format check that's failing.

Comment (saad-ali) [reply mount/mount_linux.go]: > The bug report had the error past the fs format check, so it's not the fs format check that's failing.

@gnufied can you confirm?

Comment (gnufied) [reply mount/mount_linux.go]: which bug report? https://github.com/kubernetes/utils/issues/141 or https://github.com/kubernetes/utils/issues/125 ? I assume #141 - yes the problem isn't format checking but problem was that repairing the disk actually prevented volume from being mounted and recovery (although in some cases XFS volume can't recover itself and requires `xfs_repair`).

So, the only valid reason I did not do a full revert is because I liked not having to do a `fsck` if disk is unformatted.

Comment (saad-ali) [reply mount/mount_linux.go]: Ack. SGTM
